x-pack/plugins/observability_solution/inventory/common/utils/entity_type_guards.ts

```diff
@@ -0,0 +1,25 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import type { AgentName } from '@kbn/elastic-agent-utils';
+import type { InventoryEntityLatest } from '../entities';
+
+interface EntityMap {
```

Comment:
It's not necessarily meant to scale, but to provide the necessary type guards to be able to work with `host`, `container` and `services`. Mostly because we need to access some metadata fields from these entity types manually. Is either that or `(entity as any).service?.name`, which isn't safe.

Reply:
for now, it'll work because we only show built-in entities, but when we show everything anyone could define a service entity and doesn't add a service.name field.

Reply:
Ah, we just need the `environment` here, btw. I'll remove the `name` attribute`.

>  it'll work because we only show built-in entities
IMHO that's fine. We need to get to a point where we don't need to manually access metadata attributes in the UI.